cli/internal/spinner/spinner.go

```diff
@@ -0,0 +1,81 @@
+package spinner
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/mitchellh/cli"
+	progressbar "github.com/schollz/progressbar/v3"
+	"github.com/vercel/turborepo/cli/internal/ui"
+)
+
+// getWriterAndColor unwraps cli.Ui instances until it gets to a BasicUi.
+// If it happens to spot a ColoredUi along the way, it marks that color is
+// enabled.
+func getWriterAndColor(terminal cli.Ui, useColor bool) (io.Writer, bool) {
+	switch terminal := terminal.(type) {
+	case *cli.BasicUi:
+		return terminal.Writer, useColor
+	case *cli.ColoredUi:
+		return getWriterAndColor(terminal.Ui, true)
+	case *cli.ConcurrentUi:
+		return getWriterAndColor(terminal.Ui, useColor)
+	case *cli.PrefixedUi:
+		return getWriterAndColor(terminal.Ui, useColor)
+	case *cli.MockUi:
+		return terminal.OutputWriter, false
+	default:
+		panic(fmt.Sprintf("unknown Ui: %v", terminal))
+	}
+}
+
+// WaitFor runs fn, and prints msg to the terminal if it takes longer
+// than initialDay to do complete. Depending on the terminal configuration, it may
```

Comment:
```suggestion
// than initialDelay to complete. Depending on the terminal configuration, it may
```

Reply:
Kodiak beat me here - addressed in https://github.com/vercel/turborepo/pull/1697